main.go

```diff
@@ -19,14 +19,18 @@ import (
 
 	"github.com/ericchiang/k8s"
 	corev1 "github.com/ericchiang/k8s/apis/core/v1"
+	metav1 "github.com/ericchiang/k8s/apis/meta/v1"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
-const labelNodeCompactorEnabled = "estafette.io/node-compactor-enabled"
-const labelNodeCompactorScaleDownCPURequestRatioLimit = "estafette.io/node-compactor-scale-down-cpu-request-ratio-limit"
-const labelNodeCompactorScaleDownRequiredUnderutilizedNodeCount = "estafette.io/node-compactor-scale-down-required-underutilized-node-count"
-const labelNodeCompactorScaleDownInProgress = "estafette.io/node-compactor-scale-down-in-progress"
+const (
+	labelNodeCompactorEnabled                                 = "estafette.io/node-compactor-enabled"
+	labelNodeCompactorScaleDownCPURequestRatioLimit           = "estafette.io/node-compactor-scale-down-cpu-request-ratio-limit"
+	labelNodeCompactorScaleDownRequiredUnderutilizedNodeCount = "estafette.io/node-compactor-scale-down-required-underutilized-node-count"
```

Comment:
What i've done in the other estafette controllers is introduce a 'state' annotation and json serialize a state object into it. It seems that at least `NodeCompactorScaleDownInProgress` is what you would call state, right?

Reply:
Yep, that's the only field I'm saving as state.  
I agree it's more extensible to serialize a state object even though we have only one field at the moment. I'll change it accordingly.